core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -477,7 +479,7 @@ public String toString() {
 
   private void validateRequiredInputs(Map<String, Schema> inputSchemas, FailureCollector collector) {
     for (String requiredInput : conf.getInputs()) {
-      if (!inputSchemas.containsKey(requiredInput)) {
+      if (!inputSchemas.isEmpty() && !inputSchemas.containsKey(requiredInput)) {
```

Comment:
`!inputSchemas.isEmpty() ` - this is same check as `inputSchemasAvailable` right?

Reply:
It's a bit different: `inputSchemasAvailable` returns true if:

1. `inputSchemas` is not empty, and
2. None of the values (i.e. schemas) in `inputSchemas` is null.

Reply:
I think we can use inputSchemasAvailable here like you suggested. will fix.